tests/integration/test_lock.py

```diff
@@ -348,6 +347,28 @@ def test_lock_editable_vcs_without_install(PipenvInstance, pypi):
         assert c.return_code == 0
 
 
+@pytest.mark.extras
+@pytest.mark.lock
+@pytest.mark.vcs
+@pytest.mark.needs_internet
+def test_lock_editable_vcs_with_extras_without_install(PipenvInstance, pypi):
+    with PipenvInstance(pypi=pypi, chdir=True) as p:
+        with open(p.pipfile_path, 'w') as f:
+            f.write("""
+[packages]
+requests = {git = "https://github.com/requests/requests.git", editable = true, extras = ["security"]}
+            """.strip())
+        c = p.pipenv('lock')
+        assert c.return_code == 0
+        assert 'requests' in p.lockfile['default']
+        assert 'idna' in p.lockfile['default']
+        assert 'chardet' in p.lockfile['default']
+        assert 'cryptography' in p.lockfile['default']
+        assert 'pyOpenSSL' in p.lockfile['default']
+        c = p.pipenv('install')
+        assert c.return_code == 0
```

Comment:
This is largely a copy of `test_lock_editable_vcs_without_install` above, adding the `extras` mark, swapping the `ref` for `extras` in the spec, and adding checks for the security packages.